### DISCUSSION TITLE
Using clvk vulkan opencl

### DISCUSSION THREAD

**Discussion by ffwqo** [pyopencl/pyopencl#674] [Q&A] [answered]
Is there a way to get clvk https://github.com/kpet/clvk working with pyopencl on linux? I tried copying the opencl libraries clvk generates to /etc/OpenCL/vendor and ~/.local/etc/OpencL/vendor and added the libraries to my LD_LIBRARY_PATH but neither helped finding it via cl.get_platforms() I get the usual platform error. Clinfo finds clvk and normal C programs work as well.
```
Number of platforms                               1
  Platform Name                                   clvk
  Platform Vendor                                 clvk
  Platform Version                                OpenCL 3.0 clvk
  Platform Profile                                FULL_PROFILE
  Platform Extensions                             cl_khr_icd cl_khr_extended_versioning 
  Platform Extensions with Version                cl_khr_icd                                                       0x400000 (1.0.0)
                                                  cl_khr_extended_versioning                                       0x400000 (1.0.0)
  Platform Numeric Version                        0xc00000 (3.0.0)
  Platform Extensions function suffix             clvk
  Platform Host timer resolution                  1ns

  Platform Name                                   clvk
```

**Comment by inducer** [comment, marked as answer]
If `clinfo` can find it, so should pyopencl. Just make sure both link against the same `libOpenCL.so.1`. (Use `ldd` on either to find out which that is.)  Note that `/etc/OpenCL/vendors` does not contain binaries, it contains text files with names ending in `.icd` containing file names of the shared objects with the drivers.

**Comment by ffwqo** [comment]
that worked thanks I added clvk.icd to /etc/OpenCL/vendors with libOpenCL.so in it.
